org/mozilla/jss/ssl/javax/JSSSocket.java

```diff
@@ -234,10 +234,6 @@ public JSSEngine getEngine() {
      * @see java.net.Socket#getChannel()
      */
     public JSSSocketChannel getChannel() {
-        if (parent.getChannel() == null) {
```

Comment:
I wouldn't do this either.

We've discussed this with @edewata earlier. See #532, and https://github.com/dogtagpki/jss/pull/532#discussion_r421152265.

As I understand more about this architecture... a Socket is either a blocking or non-blocking file descriptor, but a SocketChannel is non-blocking. Only in the event that a Socket is blocking will a SocketChannel be created. 

In JSS we abuse this slightly. We use the `JSSSocket` as the front to SSL operations, but in order to "do" SSL, we use the `JSSSocketChannel` as an intermediary between the `JSSSocket` and the `JSSEngine`.

Note that callers using a `SocketChannel` are mostly expected to do SSL operations themselves directly with a `SSLEngine`. There's no relevant methods in the standard to upcast a `SocketChannel` into a `SSLSocketChannel`.

Reply:
(IOW, it is tempting to grab the underlying `JSSSocketChannel`, but if you truly want non-blocking sockets, you should create a non-blocking socket to begin with...)